packages/synapse-react-client/src/components/CardContainer/CardContainer.tsx

```diff
@@ -84,22 +99,26 @@ export function CardContainer(props: CardContainerProps) {
         {isLoading && type !== OBSERVATION_CARD && loadingScreen}
       </div>
     )
-  } else if (data && data.queryResult!.queryResults.rows.length === 0) {
+  } else if (dataRows.length === 0) {
     // Show "no results" UI (see PORTALS-1497)
     return <NoContentPlaceholder />
   }
   const schema: Record<string, number> = {}
   data.queryResult!.queryResults.headers.forEach((element, index) => {
     schema[element.name] = index
   })
-  const showViewMoreButton = hasNextPage && (
+  const showViewMoreButton = (applyInitialLimit || hasNextPage) && (
```

Comment:
Show the 'View More' button if we've limited the number of results shown initially or there's another page to show